Source/Scene/ShadowMap.js

```diff
@@ -64,33 +83,70 @@ define([
         OrthographicFrustum,
         Pass,
         PerInstanceColorAppearance,
+        PerspectiveFrustum,
         Primitive) {
     'use strict';
 
-    // TODO : doc
     /**
      * @private
      */
-    function ShadowMap(context, camera) {
+    function ShadowMap(context, lightCamera) {
         this.enabled = false;
 
+        // Uniforms
+        this._shadowMapMatrix = new Matrix4();
+        this._shadowMapTexture = undefined;
+
+        this._framebuffer = undefined;
+        this._shadowMapSize = 1024;
+
+        this._lightCamera = lightCamera;
+        this._shadowMapCamera = new ShadowMapCamera();
+        this._sceneCamera = undefined;
+        this._farPlane = 100.0; // Limit the far plane of the scene camera
+        this._fitToScene = true;
+
+        var numberOfCascades = 4;
```

Comment:
We don't need to do this now, but we can consider exposing this as an option to the user if it is hard for us to set a good default given the wide array of Cesium scenes.